homeassistant/components/homekit/type_cameras.py

```diff
@@ -0,0 +1,422 @@
+    "-b:v {v_max_bitrate}k -bufsize {v_bufsize}k -maxrate {v_max_bitrate}k "
+    "-payload_type 99 "
+    "-ssrc {v_ssrc} -f rtp "
+    "-srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params {v_srtp_key} "
+    "srtp://{address}:{v_port}?rtcpport={v_port}&"
+    "localrtcpport={v_port}&pkt_size={v_pkt_size}"
+)
+
+AUDIO_ENCODER_OPUS = "libopus -application lowdelay"
+
+AUDIO_ENCODER_AAC = "libfdk_aac -profile:a aac_eld -flags +global_header"
+
+AUDIO_OUTPUT = (
+    "-map {a_map} -vn "
+    "-c:a {a_encoder} "
+    "-ac 1 -ar {a_sample_rate}k "
+    "-b:a {a_max_bitrate}k -bufsize {a_bufsize}k "
+    "-payload_type 110 "
+    "-ssrc {a_ssrc} -f rtp "
+    "-srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params {a_srtp_key} "
+    "srtp://{address}:{a_port}?rtcpport={a_port}&"
+    "localrtcpport={a_port}&pkt_size={a_pkt_size}"
+)
+
+SLOW_RESOLUTIONS = [
+    (320, 180, 15),
+    (320, 240, 15),
+]
+
+RESOLUTIONS = [
+    (320, 180),
+    (320, 240),
+    (480, 270),
+    (480, 360),
+    (640, 360),
+    (640, 480),
+    (1024, 576),
+    (1024, 768),
+    (1280, 720),
+    (1280, 960),
+    (1920, 1080),
+]
+
+VIDEO_PROFILE_NAMES = ["baseline", "main", "high"]
+
+
+@TYPES.register("Camera")
+class Camera(HomeAccessory):
+    """Generate a Camera accessory."""
+
+    def __init__(self, *args):
+        """Initialize a Camera accessory object."""
+        super().__init__(*args, category=CATEGORY_CAMERA)
+        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+        self._camera = self.hass.data[DOMAIN_CAMERA]
+
+        self.config = CAMERA_SCHEMA(self.config)
+
+        max_fps = self.config[CONF_MAX_FPS]
+        max_width = self.config[CONF_MAX_WIDTH]
+        max_height = self.config[CONF_MAX_HEIGHT]
+        resolutions = [
+            (w, h, fps)
+            for w, h, fps in SLOW_RESOLUTIONS
+            if w <= max_width and h <= max_height and fps < max_fps
+        ] + [
+            (w, h, max_fps)
+            for w, h in RESOLUTIONS
+            if w <= max_width and h <= max_height
+        ]
+
+        video_options = {
+            "codec": {
+                "profiles": [
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["BASELINE"],
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["MAIN"],
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["HIGH"],
+                ],
+                "levels": [
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE3_1"],
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE3_2"],
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE4_0"],
+                ],
+            },
+            "resolutions": resolutions,
+        }
+        audio_options = {
+            "codecs": [
+                {"type": "OPUS", "samplerate": 24},
+                {"type": "AAC-eld", "samplerate": 16},
+            ]
+        }
+
+        self.streaming_status = STREAMING_STATUS["AVAILABLE"]
```

Comment:
This is nearly a dupe of https://github.com/ikalchev/HAP-python/blob/dev/pyhap/camera.py#L411. Can we find a way to use it? Main issue is adding a flag to upstream so that SERV_MICROPHONE is optional.